README.md

```diff
@@ -2,22 +2,20 @@
 
 These packages provide utilities for generating Dart code.
 
-## Build
-
-### Interfaces
+## `build`
 
 Defines the basic pieces of how a build happens and how they interact.
```

Comment:
Lets add some more details in each of these sections, specifically around how dependencies should be set up:

## build

Defines the interfaces for creating a `Builder` which is a way of doing codegen that is compatible across build systems (pub, bazel, standalone runner, etc).

In general core packages that want to do code generation should only have a public dependency on this package (although they may have a dev dependency on one or more of the other packages).

It is recommended that transformers are exposed via a separate package, which has a dependency on `build_barback`. This allows core packages to not leak the dependency on `barback` and all of its transitive deps to all consumers.

## build_runner

Provides `build` and `watch` utilities to enact builds.

This package generally only be a `dev_dependency` as it is used to run actual standalone builds. The only exception to that would be if somebody was wrapping the `build` and `watch` methods with some other package.


## build_test

Stub implementations for classes in `Build` and some utilities for running instances of builds and checking their outputs.

This package generally only be a `dev_dependency` as it introduces a dependency on `package:test`. The exception to that would be if you were creating another testing-only package that wraps this one.